transaction/core/src/mint/tx.rs

```diff
@@ -17,8 +17,8 @@ use serde::{Deserialize, Serialize};
 )]
 pub struct MintTxPrefix {
     /// Token ID we are minting.
-    #[prost(uint32, tag = "1")]
-    pub token_id: u32,
+    #[prost(uint64, tag = "1")]
+    pub token_id: u64,
```

Comment:
How about we make this a `TokenId`? (with the Prost tag specifying uint64)

```suggestion
    pub token_id: TokenId,
```

Reply:
Have you tested that this compiles?